### DISCUSSION TITLE
Remove or rename Listing caption

### DISCUSSION THREAD

**Discussion by vfacta** [quarto-dev/quarto-cli#5510] [Q&A]
Considering the following code block in a .qmd file:

````
```{.python filename="helllo.py" }
print("hello, world!")
```
````

 It is converted in Latex's pdf output to something like this:

```
Listing 1 helllo.py

print("hello, world!")
```

I have already tried including in header the following commands, but nothing changed:

```
\usepackage{listings}
\renewcommand{\lstlistingname}{}
```

Here is a minimal working example:

````
---
title: "Untitled"
format: 
    pdf:
      include-in-header: 
        text: |
          \usepackage{caption}
          \usepackage{listings}
          \renewcommand{\lstlistingname}{}
---

```{.python filename="helllo.py"}
print("hello, world!")
```
````

How can I rename the listing caption or remove it entirely with its number?

**Comment by mcanouil** [comment]
Could you share a small "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Reply by vfacta** [reply]
sure, just updated the post

**Comment by vfacta** [comment]
Update: Following intructions from [Quarto cross-references doc](https://quarto.org/docs/authoring/cross-references.html#titles), I managed to remove the name "Listing" adding an empty string `''` to `lst-title` in `crossref`:

```
crossref:
  lst-title: ''
```
However, it doesn't seem to be an option for removing the listing's number

**Reply by cderv** [reply]
Just to be sure - you are looking to only have `hello.py` shown in the box header above the code ? and remove "Listing 1" ?,

**Reply by vfacta** [reply]
Yes, that's the idea

**Reply by cderv** [reply]
Unfortunatly I don't think you can do that. Those environment are numbered by definition, and I don't think you can unnumbered them. You could try not use the listing feature but another LaTeX tools that allows you to create a box with a caption and that you can tweak how you like it

**Comment by mcanouil** [comment]
You might need to use LaTeX partials here, see <https://quarto.org/docs/journals/templates.html#latex-partials>.
In particular, the whole `template.tex` (it's where `listings` LaTeX package is imported, thus why your command did not change anything).
